### PR TITLE
Restaurant design and functionality for Assignment 1

### PR DESCRIPTION
**Pull request by Sunzi2598**
The Pull Request contains the functionality of the Food Ordering System for Assignment 1.
The HTML structure is in index.html
The CSS design with all styles is in style.css
slider.js contains the code for the price slider, imageHover.js contains the code for image effects and buttonAdd.js contains the implementation of the dashboard functionality for adding a new product.
The link of the GitHub pages is: https://sunzi2598.github.io/Sunzi-Bistro/

### CODE REVIEW

File: buttonAdd.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    name: document.getElementById("name").value,
+    category: document.getElementById("category").value,
+    price: document.getElementById("price").value,
+  };
+  data.push(product);
+  // document.forms[0].reset(); // to clear the form for the next entries
+  document.querySelector("form").reset();
+
+  //for display purposes only
+  console.warn("added", { data });
```
Comment (AdrianFlorian) [inline buttonAdd.js]: Never commit with console logs :-), those are for you to see while developing, but they should never go to master :-)

File: buttonAdd.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+  //saving to localStorage
```
Comment (AdrianFlorian) [inline buttonAdd.js]: whenever you feel the need to add a comment above a block of code, it means that you should put it in a new method/function. A good code should be self documented and should not need comments :-) - in this example I would create a new method `saveToLocalStorage` and call it - makes it pretty self explanatory, right?

File: buttonAdd.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  localStorage.setItem("MyProductList", JSON.stringify(data));
+  addToTable();
+};
+document.addEventListener("DOMContentLoaded", () => {
+  document.getElementById("btnAdd").addEventListener("click", addProduct);
+});
+
+// function addToTable() {
```
Comment (AdrianFlorian) [inline buttonAdd.js]: If you don't need it, remove it :-)

File: buttonAdd.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+//     let r3 = tabel.insertRow();
+//     let r4 = tabel.insertRow();
+//     r1.innerHTML = data[i].photo;
+//     r2.innerHTML = data[i].name;
+//     r3.innerHTML = data[i].category;
+//     r4.innerHTML = data[i].price;
+//   }
+// }
+
+function addToTable() {
```
Comment (AdrianFlorian) [inline buttonAdd.js]: Add what to table? Remember the comment above, the method names should be very clear :-)

File: imageHover.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+function bigImg(x) {
+  x.style.height = "250px";
```
Comment (AdrianFlorian) [inline imageHover.js]: Method names should signify if they are altering the state of an object or if they are just returning an object. In this case, those 2 methods are altering the height and width of the image and it should be clear, something like `makeImageBig` or `applyBigImageStyling`. Also do not use variable names like `x` use a self explanatory name instead, like `image`

**Review by AdrianFlorian — CHANGES_REQUESTED** [review]
Nice job Alexandra, 👏, just some overall comments:
- be careful on the method naming and comments
- keep the code as clean as possible, without any commented out code (even if you think you might need it)
- please rename the pull request name and description to reflect the actual changes

File: buttonAdd.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+  document.querySelector(".bg-modal").style.display = "flex";
+});
+
+document.querySelector(".close").addEventListener("click", function () {
+  document.querySelector(".bg-modal").style.display = "none";
+});
+
+let data = [];
+const addProduct = (ev) => {
+  ev.preventDefault(); //to stop the form submitting
```
Comment (chioreandan) [inline buttonAdd.js]: No need for this kind of comments, it should be clear from the code :D

Comment (chioreandan) [reply buttonAdd.js]: I know that the teachers say that it is a good practice to add comments but in real life, it is not. The code naming of classes, methods, etc should make the code readable and easy to understand.

File: index.html [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+            <br>
+            <a href="#" class="button" id="btnAdd">Add</a>
+          </form>
+        </div>
+      </div>
+      <table class="image-table">
+        <tbody>
+          <tr class="statistics">
+          <td>
+            <img class="image" onmouseover="bigImg(this)" onmouseout="normalImg(this)" src="img/Pizza Salami.webp" alt="Pizza Salami" title="Pizza Salami">
```
Comment (chioreandan) [inline index.html]: Pretty hard to read such a long line. When you have so many attributes you can put 1 bellow another.

File: style.css [hunk 8]
```diff
@@ -0,0 +1,478 @@
...
+  text-align: left;
+  margin-left: -75px;
+  margin-top: -18px;
+  font-size: 15px;
+  color: rgba(0, 0, 0, 0.404);
+  font-weight: bold;
+}
+
+/* Product */
+.image {
```
Comment (chioreandan) [inline style.css]: I usually name this kind of classes .image-wrapper

File: style.css [hunk 9]
```diff
@@ -0,0 +1,478 @@
...
+  height: 170px;
+  width: auto;
+  margin-top: 0px;
+  margin-left: 60px;
+  border-radius: 15px;
+}
+
+.image1 {
```
Comment (chioreandan) [inline style.css]: This is not a good naming for this class. Avoid using classes and methods like this image1, method1, method3, etc

File: style.css [hunk 10]
```diff
@@ -0,0 +1,478 @@
...
+
+h4 {
+  font-size: 12px;
+  margin-top: -10px;
+  margin-bottom: 10px;
+  color: rgb(236, 138, 68);
+  font-weight: bold;
+}
+
+.button2 {
```
Comment (chioreandan) [inline style.css]: same here, naming

**Review by chioreandan — CHANGES_REQUESTED** [review]
Good Job @Sunzi2598,
Take more care when naming classes and methods and try to use less comments.

File: buttonAdd.js [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+  ev.preventDefault();
+  let product = {
+    id: Date.now(),
+    photo: document.getElementById("photo").value,
+    name: document.getElementById("name").value,
+    category: document.getElementById("category").value,
+    price: document.getElementById("price").value,
+  };
+  data.push(product);
+  // document.forms[0].reset(); // to clear the form for the next entries
```
Comment (AdrianFlorian) [inline buttonAdd.js]: do you need this one? :-)

File: buttonAdd.js [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+  document.querySelector("form").reset();
+
+  saveToLocalStorage();
+  addProductToTable();
+};
+document.addEventListener("DOMContentLoaded", () => {
+  document.getElementById("btnAdd").addEventListener("click", addProduct);
+});
+
+//saving to localStorage
```
Comment (AdrianFlorian) [inline buttonAdd.js]: why do you need this comment?

File: buttonAdd.js [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+function saveToLocalStorage() {
+  localStorage.setItem("MyProductList", JSON.stringify(data));
+}
+
+//add a new product
```
Comment (AdrianFlorian) [inline buttonAdd.js]: same here, why need the comment? :-)

**Review by AdrianFlorian — CHANGES_REQUESTED** [review]

**Comment by AdrianFlorian** [comment]
@Sunzi2598 can you please review your own PR and let me know what can be impoved based on the previous comments? :-)

**Comment by Sunzi2598** [comment]
I have to change the name for dropdown containers and buttons

**Comment by Sunzi2598** [comment]
@AdrianFlorian

**Comment by AdrianFlorian** [comment]
@Sunzi2598 what about the spacing/newlines? :-)

**Comment by Sunzi2598** [comment]
@AdrianFlorian @chioreandan

File: buttonAdd.js [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+});
+
+function saveToLocalStorage() {
+  localStorage.setItem("MyProductList", JSON.stringify(data));
+}
+
+function addProductToTable() {
+  let tabel = document.querySelector("table");
+  let row = tabel.insertRow();
+  for (let i = 0; i < data.length; i++) {
```
Comment (chioreandan) [inline buttonAdd.js]: Use ``` forEach() ``` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach

File: buttonAdd.js [hunk 15]
```diff
@@ -0,0 +1,55 @@
...
+    category: document.getElementById("category").value,
+    price: document.getElementById("price").value,
+  };
+
+  data.push(product);
+  document.querySelector("form").reset();
+
+  saveToLocalStorage();
+  addProductToTable();
+};
```
Comment (chioreandan) [inline buttonAdd.js:26]: add empty line

File: index.html [hunk 16]
```diff
@@ -0,0 +1,215 @@
...
+    <title>Sunzi Bistro</title>
+  </head>
+  <body>
+    <header>
+        <nav>
+            <div class = "nav" >
+                <img src = "img/logo.jpg" alt = "Sunzi Bistro logo" class = "logo" >
+                <h3> &nbsp; Sunzi <br> &nbsp; Bistro </h3>
+                <ul class = "main-nav" >
+                    <li><a href = "#home" >Home</a></li>
```
Comment (chioreandan) [inline index.html]: @Sunzi2598 you don't need spaces between arguments and value. It should look something like this
 ``` class="main-nav" ```

Comment (chioreandan) [reply index.html]: Please apply this change everywhere.

File: index.html [hunk 17]
```diff
@@ -0,0 +1,215 @@
...
+      content = "width=device-width, initial-scale = 1, minimum-scale = 1"
+    />
+    <link rel = "stylesheet" 
+          href = "https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.3/css/all.min.css" 
+          integrity = "sha512-iBBXm8fW90+nuLcSKlbmrPcLa0OT92xO1BIsZ+ywDWZCvqsWgccV3gFoRBv0z+8dLJgyAHIhR35VZc2oM/gI1w==" 
+          crossorigin = "anonymous" 
+    />
+    <link rel = "stylesheet" href = "style.css" />
+    <title>Sunzi Bistro</title>
+  </head>
```
Comment (chioreandan) [inline index.html:16]: add empty line after each big section/block

File: slider.js [hunk 18]
```diff
@@ -0,0 +1,126 @@
+let thumbsize = 14;
+
+function draw(slider, splitvalue) {
+  /* set function vars */
```
Comment (chioreandan) [inline slider.js]: I think anybody that heard of programming knows what this section does. Please remove all your unnecessary comments as @AdrianFlorian said.

**Review by chioreandan — CHANGES_REQUESTED** [review]

**Comment by Sunzi2598** [comment]
Can you check again please? @AdrianFlorian @chioreandan

**Comment by AdrianFlorian** [comment]
@Sunzi2598 https://chris.beams.io/posts/git-commit/ please go through this - it's important to follow those guidelines for the Git commit messages as well as the branching naming conventions :-)

File: buttonAdd.js [hunk 19]
```diff
@@ -0,0 +1,56 @@
...
+
+document.addEventListener("DOMContentLoaded", () => {
+  document.getElementById("btnAdd").addEventListener("click", addProduct);
+});
+
+function saveToLocalStorage() {
+  localStorage.setItem("MyProductList", JSON.stringify(data));
+}
+
+function addProductToTable() {
```
Comment (AdrianFlorian) [inline buttonAdd.js:36]: nice 👍🏻. Isn't this much nicer, @Sunzi2598 ? :-)

**Review by AdrianFlorian — APPROVED** [review]
👏 nice job Alexandra!

**Review by chioreandan — APPROVED** [review]
Looking good. Try to also change the tab spacing for `index.html` to 2 spaces.

**Pull request merged**